dask/dataframe/core.py

```diff
@@ -1461,17 +1461,76 @@ def var(self, axis=None, skipna=True, ddof=1, split_every=False, dtype=None, out
                                     axis=axis, skipna=skipna, ddof=ddof)
             return handle_out(out, result)
         else:
-            num = self._get_numeric_data()
-            x = 1.0 * num.sum(skipna=skipna, split_every=split_every)
-            x2 = 1.0 * (num ** 2).sum(skipna=skipna, split_every=split_every)
-            n = num.count(split_every=split_every)
-            name = self._token_prefix + 'var'
-            result = map_partitions(methods.var_aggregate, x2, x, n,
-                                    token=name, meta=meta, ddof=ddof)
+            if self.ndim == 1:
+                result = self._var_1d(self, skipna, ddof, split_every)
+            elif len(self._meta_nonempty.select_dtypes(include=[np.timedelta64]).columns) > 0:
+                result = self._var_mixed(skipna, ddof, split_every)
+            else:
+                result = self._var_numeric(skipna, ddof, split_every)
+
             if isinstance(self, DataFrame):
                 result.divisions = (min(self.columns), max(self.columns))
             return handle_out(out, result)
 
+    def _var_numeric(self, skipna=True, ddof=1, split_every=False):
+        num = self.select_dtypes(include=['number', 'bool'])
+
+        values_dtype = num.values.dtype
+        array_values = num.values
+
+        if not np.issubdtype(values_dtype, np.number):
+            array_values = num.values.astype('f8')
+
+        var = da.nanvar if skipna or skipna is None else da.var
```

Comment:
I don't think that `skipna=None` should be interpreted as `True`. I would think `var = da.nanvar if skipna else da.var`.

Reply:
@TomAugspurger I would also say this is arguable but this is a behavior of pandas:

```python 
df = pd.DataFrame({"A": [1.0, 1.2, 3.6, np.nan], "B": pd.to_timedelta([1, 2, 3, np.nan])})
df.var(skipna=None)
A    2.093333
B    1.000000
dtype: float64
df.var(skipna=False)
A   NaN
B   NaN
dtype: float64
df.var(skipna=True)
A    2.093333
B    1.000000
dtype: float64
df.var()
A    2.093333
B    1.000000
dtype: float64
```

Reply:
Well that's an unfortunate design :) Sorry about that, let's leave it to match pandas.